source/extensions/listener_managers/listener_manager/listener_impl.cc

```diff
@@ -592,6 +592,7 @@ void ListenerImpl::buildUdpListenerWorkerRouter(const Network::Address::Instance
   }
   udp_listener_config_->listener_worker_routers_.emplace(
       address.asString(), std::make_unique<Network::UdpListenerWorkerRouterImpl>(concurrency));
+  parent_.server_.hotRestart().registerUdpForwardingListener(address, *udp_listener_config_);
```

Comment:
Do think it make sense to call registerUdpForwardingListener() outside of this method where you have access to the address shared_ptr?

Reply:
I don't think it's necessary/useful since we have no choice but to index the map by string anyway (like the line above also does!)
(Rather, there is actually a choice, but if we were to index with an address and a custom hash function then the same treatment could be given to listener_worker_routers_ and it would turn into a big refactor that I'd very much rather not do at this time.)

Reply:
Done.